src/shapes/Object/InteractiveObject.ts

```diff
@@ -462,7 +462,10 @@ export class InteractiveFabricObject<
               // and is the qrDecompose of a matrix that takes in account zoom too
               new Point(options.scaleX, options.scaleY)
         ).scalarMultiply(this.strokeWidth);
-      size = bbox.add(stroke).scalarAdd(this.borderScaleFactor);
+      size = bbox
+        .add(stroke)
+        .scalarAdd(this.borderScaleFactor)
+        .scalarAdd(this.padding * 2);
```

Comment:
padding does not change with zoom. Is another PITA like stroke uniform.
Which is the CTX state here? is it multiplied already by the viewport transform? or is it scaled back to neutral?

Can you please add a visual test like you did but with also a bunch of zoom?

Reply:
i will publish another release after this fix.

Reply:
Also, just to be clear, padding shouldn't enter in the layout of the group, so if an object inside the group has padding 2000, it will blow out of the layout when selected and that is fine.

Reply:
I have added a test with scaling, did you look at it?
It does not affect group layout

Reply:
> Which is the CTX state here? 

Same as for rendering controls

Reply:
actually I have to say I think we should use the coords to render the border but that is not related to here

Reply:
Ahh I missed vpt, sorry